configure

```diff
@@ -467,6 +467,26 @@ else
    die "Make sure it is installed and available in your PATH."
 fi
 
+dir_separator=$("$LUA_BINDIR/$LUA_INTERPRETER" -e 'print(package.config:sub(1, 1))')
+if [ "$dir_separator" = "\\" ]
+then
+  echo "Lua platform: $(GREEN "Mingw-w64 on MSYS2")"
+  LUA_MINGW_W64_MSYS2=yes
+  LUA_DIR="$(cygpath --windows "$LUA_DIR" | sed -e 's/\\/\\\\/g')"
+  LUA_BINDIR="$(cygpath --windows "$LUA_BINDIR" | sed -e 's/\\/\\\\/g')"
+  if [ "$LUA_INCDIR_SET" = "yes" ]
+  then
+    LUA_INCDIR="$(cygpath --windows "$LUA_INCDIR" | sed -e 's/\\/\\\\/g')"
+  fi
+  if [ "$LUA_LIBDIR_SET" = "yes" ]
+  then
+    LUA_LIBDIR="$(cygpath --windows "$LUA_LIBDIR" | sed -e 's/\\/\\\\/g')"
+  fi
+else
+  echo "Lua platform: $(GREEN "not Mingw-w64 on MSYS2")"
+  LUA_MINGW_W64_MSYS2=no
+fi
```

Comment:
MSYS2 + Mingw_w64 needs Windows style path. So we detect MSYS2 + Mingw_w64 here.